### DISCUSSION TITLE
Popularity of Transcrypt

### DISCUSSION THREAD

**Discussion by halloleo** [QQuick/Transcrypt#833] [General]
I thought I start a first discussion here....

Currently I am rediscovering Transcrypt:   2 years agoI had used it for parts of aweb project and now I embark on another (small) Transcrypt  project. I like how clean the generated JavaScript code is: You can easily use Transcrypt as part of a bigger JS project. &ndash; This is something you cannot say from other to-JS transpilers (I am thinking NImJS or ClojureJS). They all work well, but the JavaScript they produce is essentially unreadable. &ndash; Here's to Transcrypt! As its motto says: Transcrypt is "Lean, fast, open" &ndash; and Python! But Transcrypt is not super popular. The community seems to be _very_ small.

So I want to ask: _Why's that -- and can we do something about it?_

**Comment by JennaSys** [comment]
Great topic!

For one, I don't think many Python developers even know about Transcrypt. I always try to bring it up in conversations when it makes sense to do so, and I have published blog posts related to its use (or more specifically, how I use it).

The second is, even for people that have heard about it, I think many don't actually understand how it would be used. And like Python itself, there are many way it can be utilized. Or, they are quick to apply the adage "use the right tool for the job" without considering nuances of that statement like already having proficiency in the "right tool", or the overall developer experience.

The way I use Transcrypt is pretty straightforward and focused - I use Python to create React web applications. But just looking at the demos here, I see that there are ways to use it that don't necessarily line up with the way I'm using it. So I do try to be open minded about other use cases for Transcrypt too.

I'm in the process of putting together a talk that demonstrates how I use Transcrypt. Some of the points I want to specifically try and make with the talk include:
- Show that the tech stack isn't messy, is easy to use, and has minimal setup requirements
- Show that clean idiomatic Python code is possible, and that Python constructs (dictionaries, list comprehensions, lambdas, f-strings, etc.) are all usable.
- Show that the code can be all Python and that there is no need for intermixing languages in a module or using weird syntax
- Show that attractive UIs can be easily achieved (with 3rd party styled React components)
- Show HMR capability (with the Parcel bundler) and debugging Python code in the browser thanks to source mapping
- Show that the generated JS is readable if you ever need to dig into it

I'll do the talk at local Python user groups, possibly do a YouTube video, and eventually try to get in as a speaker at other events.

I've been using Transcrypt now for about 3 years and I'm amazed at well it works. Right now, I'm in the middle of working on a bigger full-stack Flask/React application, and I'm still blown away by the fact that I can write clean Python code for the entire application and never have to think about JavaScript. I would think that more Python developers would be interested in this. I've seriously never had a better experience developing web applications than I have had since discovering the existence of Transcrypt.

There are of course other "Python in the browser" options as well that might appeal to other situations. PyScript in particular has received a LOT of press over the last few months. While I can see it is definitely a boon for some types of programs, I think Transcrypt has a wider number of places it can be utilized.

**Reply by martin-braun** [reply]
Hi @JennaSys,

I think having support for Mojo source files could boost the popularity of this project. A lot of people will move to Python once Mojo established, because pure Python is too slow. While Transcrypt couldn't leverage all benefits from Mojo, it could enable types for JavaScript (and some other cool concepts from Mojo) similar how TypeScript is doing the same.

**Reply by halloleo** [reply]
Mmmh, I'm not convinced Moho support will help a lot for two reason:

1. Mojo is extremely young and (apart of some hype) not yet any real player in the Python space.
2. Mojo targets performance critical AI application, while Transcrypt is geared towards the web.

**Reply by martin-braun** [reply]
@halloleo I'm actually not talking about now, but in the future when it has established and stabilized. Mojo will not be limited to AI. Just to be clear, I would always pick Typescript over Python for web development, because of type safety. Transcrypt could bridge a gap in this realm, but that's worth a discussion way later.

**Comment by classifaddict** [comment]
Hello,

There are several tools named Mojo, which one are you talking about?

Best,
Olivier

Le mer. 30 août 2023, 15:33, Martin Braun ***@***.***> a
écrit :

**Reply by halloleo** [reply]
I reckon @martin-braun means [Mojo 🔥: Programming language for all of AI](https://www.modular.com/mojo).
